src/frequenz/sdk/config/_manager.py

```diff
@@ -80,20 +95,53 @@ def __str__(self) -> str:
         return f"{type(self).__name__}[{self.name}]"
 
     # The noqa DOC502 is needed because we raise TimeoutError indirectly.
-    async def new_receiver(self) -> Receiver[Mapping[str, Any] | None]:  # noqa: DOC502
+    async def new_receiver(  # pylint: disable=too-many-arguments # noqa: DOC502
+        self,
+        *,
+        wait_for_first: bool = False,
+    ) -> Receiver[Mapping[str, Any] | None]:
         """Create a new receiver for the configuration.
 
         Note:
             If there is a burst of configuration updates, the receiver will only
             receive the last configuration, older configurations will be ignored.
 
+        ### Waiting for the first configuration
+
+        If `wait_for_first` is `True`, the receiver will wait for the first
+        configuration to be received before returning the receiver. If the
+        configuration can't be received in time, a timeout error will be raised.
+
+        If the configuration is received successfully, the first configuration can be
+        simply retrieved by calling [`consume()`][frequenz.channels.Receiver.consume] on
+        the receiver without blocking.
+
         Example:
             ```python
             # TODO: Add Example
             ```
 
+        Args:
+            wait_for_first: Whether to wait for the first configuration to be received
+                before returning the receiver. If the configuration can't be received
+                for
+                [`wait_for_first_timeout`][frequenz.sdk.config.ConfigManager.wait_for_first_timeout]
+                time, a timeout error will be raised. If receiving was successful, the
+                first configuration can be simply retrieved by calling
+                [`consume()`][frequenz.channels.Receiver.consume] on the receiver.
+
         Returns:
             The receiver for the configuration.
+
+        Raises:
+            asyncio.TimeoutError: If `wait_for_first` is `True` and the first
+                configuration can't be received in time.
         """
         recv_name = f"{self}_receiver"
-        return self.config_channel.new_receiver(name=recv_name, limit=1)
+        receiver = self.config_channel.new_receiver(name=recv_name, limit=1)
+
+        if wait_for_first:
+            async with asyncio.timeout(self.wait_for_first_timeout.total_seconds()):
+                await receiver.ready()
+
+        return receiver
```

Comment:
Did you test if it will work if actor receive timeout, print error log and call this method again.
Can we be sure that previous receiver will be deallocated and it will work correctly (no name conflict)?

Reply:
Good question. The old receiver should be eventually `del`eted by the GC. When we have the [`close`](https://github.com/frequenz-floss/frequenz-channels-python/pull/348) method in receivers, we can make it more explicit (we will need to fix this in **many** places).

Reusing the name could be confusing if both the old and new receiver are active for some time, but it shouldn't be a *problem*, the name is only used for informational purposes, it's not like they clash in any other way.